src/mlpack/methods/ann/layer/multihead_attention.hpp

```diff
@@ -77,25 +74,27 @@ class MultiheadAttentionType : public Layer<InputType, OutputType>
    * @param numHeads Number of parallel attention heads.
    * @param attnMask Two dimensional Attention Mask.
    * @param keyPaddingMask Key Padding Mask.
+   * @param selfAttention Use self-attention; source key, query, and value all come from the same inputs
    */
   MultiheadAttentionType(const size_t tgtSeqLen,
                          const size_t srcSeqLen,
                          const size_t embedDim,
                          const size_t numHeads,
-                         const InputType& attnmask = InputType(),
-                         const InputType& keyPaddingMask = InputType());
+                         const MatType& attnmask = MatType(),
+                         const MatType& keyPaddingMask = MatType(),
```

Comment:
Agree with the 0/1 thing.  Could just internally convert to -inf/0 so the math works the same (in keeping with the original papers).  Is that confusing though if people are expecting this to work as described in the papers?  Or should we give them the benefit of the doubt that they can read the documentation (assuming it makes sense).  ;)

Reply:
Also, just noticed a bit of an inconsistency.  It takes these masks in the constructor, but then it doesn't serialize/deserialize them (as if they are meant to be set explicitly during training/use, rather than during construction time.  Should I just remove them from the constructor?  Or add them to the serialization.

Reply:
I'm answering my own question and removing them.  ;)